app/Actions/Photo/Create.php

```diff
@@ -118,6 +121,22 @@ public function add(
 			$res = $this->save($this->photo);
 		}
 
+		//extract the main color and palette
+		$palette = ColorThief::getPalette($this->path, 5);
+
+		if (!empty($palette)) {
+			foreach ($palette as $index => $value) {
+				$color = new Color([
+					'r' => $value[0],
+					'g' => $value[1],
+					'b' => $value[2],
+					'is_main' => ($index === 0 ? 1 : 0),
+					'photo_id' => $this->photo->id,
+				]);
```

Comment:
You should not set the id of the foreign relation manually. `$this->photo->colors()->save($color)` (1 line below) takes care of that. Your can remove `'photo_id' => $this->photo-id'`.